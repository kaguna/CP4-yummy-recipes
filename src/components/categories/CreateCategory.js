import React, { Component } from 'react';
import axios from 'axios'
import Home from '../Home';
class CreateCategory extends Component {
    constructor(props){
        super(props);
        this.state = { category_name: ""}
    }

    inputHandler = (event) => {
        const {name, value} = event.target;
        this.setState({[name]: value});
    };

    createCategoryHandler = (event) => {
        const {category_name} = this.state;
        event.preventDefault();
        const header = {headers:{'x-access-token': window.localStorage.getItem('token')},
            content_type: 'application/json'};
        if (!header)
            window.location.href='/login';
        axios.post("http://127.0.0.1:5000/categories/", {category_name}, header)
            .then(response => {
                this.setState({mess: response.data.message, error: "",});
                this.props.categoryAfterCreation();
            })
            .catch(error => {
                if (error.response) {
                    this.setState({error: error.response.data.message, mess: ""});
                    console.log(error.response.data.message)
                } else if (error.request) {
                    this.setState({error:"Can't connect to the server.Please check your connection and try again.", mess: ""});
                }
            });
    };
    render() {
        return (
            <div className="modal fade" id="createcategory" tabindex = "-1" role="dialog"
                 aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
                <div className="modal-dialog" role="document">
                    <div className="modal-content">
                        <div class="modal-header">
                            <h5 className="modal-title" id="exampleModalLongTitle">Create category</h5>
                        </div>
                        <div className="modal-body">
                            {this.state.mess?
                            <div className="alert alert-success">{this.state.mess}</div>: ""}
                            {this.state.error?
                                <div className="alert alert-danger">{this.state.error}</div>: ""}
                            <form className="form-horizontal" onSubmit={this.createCategoryHandler}>
                                <div className="form-group">
                                    <label className="control-label col-sm-3">Category Name:</label>
                                    <div className="col-sm-9">
                                        <input type="text" name="category_name" className="form-control" placeholder="Enter Category name"
                                               value={this.state.category_name} onChange={this.inputHandler}/>
                                    </div>
                                </div>
                                <div className="modal-footer">
                                    <button type="submit" className="btn btn-danger" data-dismiss="modal">Close</button>
                                    <button type="submit" className="btn btn-primary">
                                        <i class="glyphicon glyphicon-plus"></i> Add Category
                                    </button>
                                </div>
                            </form>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
export default CreateCategory;
