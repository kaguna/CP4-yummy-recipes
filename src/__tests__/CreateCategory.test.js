import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CreateCategory from '../components/categories/CreateCategory';

describe('Create category component', () => {
  const wrapper = shallow(<CreateCategory />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
