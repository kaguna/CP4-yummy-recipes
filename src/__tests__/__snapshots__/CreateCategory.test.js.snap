// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create category component renders properly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalCenterTitle"
  className="modal fade"
  id="createcategory"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLongTitle"
        >
          Create Category
        </h5>
      </div>
      <div
        className="modal-body"
      >
        <form
          className="form-horizontal"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              className="control-label col-sm-3"
            >
              Category Name:
            </label>
            <div
              className="col-sm-9"
            >
              <input
                className="form-control"
                name="categoryName"
                onChange={[Function]}
                placeholder="Enter Category name"
                type="text"
              />
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-danger"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
            <button
              className="btn btn-primary"
              type="submit"
            >
              <i
                className="glyphicon glyphicon-plus"
              />
               Add Category
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
